### PR TITLE
[common] Share a common implementation of ZpoolName

### PR DESCRIPTION
**Pull request by smklein**
This removes dependencies on `illumos-utils`, and obviates some unnecessary parsing.

### CODE REVIEW

**Comment by smklein** [comment]
This PR is used by https://github.com/oxidecomputer/omicron/pull/5599

File: nexus/reconfigurator/execution/src/datasets.rs [hunk 1]
```diff
@@ -70,19 +69,7 @@ pub(crate) async fn ensure_crucible_dataset_records_exist(
 
         // Map progenitor client strings into the types we need. We never
         // expect these to fail.
```
Comment (jgallagher) [inline nexus/reconfigurator/execution/src/datasets.rs:71]: This comment can go away too I think? It doesn't look like we're doing any other progenitor client type conversion after this change.

File: nexus/reconfigurator/planning/src/blueprint_builder/builder.rs [hunk 2]
```diff
@@ -112,13 +111,8 @@ pub enum EnsureMultiple {
...
 }
 
-fn zpool_id_to_external_name(zpool_id: ZpoolUuid) -> anyhow::Result<ZpoolName> {
-    let pool_name_generated =
-        illumos_utils::zpool::ZpoolName::new_external(zpool_id).to_string();
-    let pool_name = ZpoolName::from_str(&pool_name_generated).map_err(|e| {
-        anyhow!("Failed to create zpool name from {zpool_id}: {e}")
-    })?;
-    Ok(pool_name)
+fn zpool_id_to_external_name(zpool_id: ZpoolUuid) -> ZpoolName {
```
Comment (jgallagher) [inline nexus/reconfigurator/planning/src/blueprint_builder/builder.rs:114]: Not sure this function is useful anymore - maybe remove and inline `ZpoolName::new_external()` everywhere?

**Review by jgallagher — APPROVED** [review]

File: nexus/reconfigurator/execution/src/datasets.rs (hunk 1 above)
Comment (smklein) [reply nexus/reconfigurator/execution/src/datasets.rs]: Done

File: nexus/reconfigurator/planning/src/blueprint_builder/builder.rs (hunk 2 above)
Comment (smklein) [reply nexus/reconfigurator/planning/src/blueprint_builder/builder.rs]: Done

**Pull request merged**